lib/ansible/modules/cloud/ovirt/ovirt_role.py

```diff
@@ -0,0 +1,177 @@
+    search_by_attributes,
+    search_by_name,
+)
+from ansible.module_utils.basic import AnsibleModule
+import traceback
+
+try:
+    import ovirtsdk4.types as otypes
+except ImportError:
+    pass
+
+
+class RoleModule(BaseModule):
+    def build_entity(self):
+        all_permits = self.get_all_permits()
+        return otypes.Role(
+            id=self.param('id'),
+            name=self.param('name'),
+            mutable=self.param('mutable') if self.param('mutable') else None,
+            administrative=self.param('administrative') if self.param(
+                'administrative') else None,
+            permits=[
+                otypes.Permit(id=all_permits.get(new_permit)) for new_permit in self.param('permits')
+            ]
+        )
+
+    def get_all_permits(self):
+        return dict((permit.name, permit.id) for permit in self._connection.system_service().cluster_levels_service().level_service('4.3').get().permits)
+
+    def update_check(self, entity):
+        def check_permits():
+            if self.param('permits'):
+                permits_service = self._service.service(entity.id).permits_service()
+                current = [er.name for er in permits_service.list()]
+                passed = [pr for pr in self.param('permits')]
+                if not sorted(current) == sorted(passed):
+                    # remove all
+                    for permit in permits_service.list():
+                        permits_service.permit_service(permit.id).remove()
+                    # add passed permits
+                    all_permits = self.get_all_permits()
+                    for new_permit in passed:
+                        permits_service.add(otypes.Permit(id=all_permits.get(new_permit)))
+                    return False
+            return True
+
+        return (
+            check_permits() and
+            equal(self.param('mutable'), entity.mutable) and
+            equal(self.param('administrative'), entity.administrative)
+        )
+
+
+def main():
+    argument_spec = ovirt_full_argument_spec(
+        state=dict(
+            choices=['present', 'absent'],
+            default='present',
+        ),
+        id=dict(default=None),
+        name=dict(default=None),
+        mutable=dict(default=True, type='bool'),
+        administrative=dict(default=False, type='bool'),
+        permits=dict(type='list', default=[]),
+    )
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        required_one_of=[['id', 'name']],
+    )
+
+    check_sdk(module)
+
+    try:
+        auth = module.params.pop('auth')
+        connection = create_connection(auth)
+        roles_service = connection.system_service().roles_service()
+        roles_module = RoleModule(
+            connection=connection,
+            module=module,
+            service=roles_service,
+        )
+        state = module.params['state']
+        if state == 'present':
+            module.params.get('permits').append('login')
```

Comment:
I just wonder what will happen if user specify login and you append login, maybe it will fail on engine side? Can you please use set and cast it to list?